dagger/rest/api.nim

```diff
@@ -86,9 +86,14 @@ proc initRestApi*(node: DaggerNodeRef): RestRouter =
             peerRecord.get().addresses.mapIt(
               it.address
             )
+      try:
+        await node.connect(peerId.get(), addresses)
+        return RestApiResponse.response("Successfully connected to peer")
+      except DialFailedError as e:
+        return RestApiResponse.error(Http400, "Unable to dial peer")
+      except CatchableError as e:
```

Comment:
You need to handle CancelledError like so:
```nim
except CancelledError as e:
  raise e
except CatchableError as e:
###
```

Reply:
Otherwise, the CancelledError will be swallowed, and cancellation will not be handled properly.
(not sure if this procedure can be cancelled though)

Reply:
Hmm, actually I wonder if that is the case since this are essentially callbacks and we don't need to teardown anything in presto itself and the best thing presto can do with the cancellation is return a `500` (or similar)?

Reply:
In fact this is how presto handles this callbacks:

```nim
        let restRes =
          try:
            await route.callback(request, pathParams, queryParams, optBody)
          except HttpCriticalError as exc:
            debug "Critical error occurred while processing a request",
                  meth = $request.meth, peer = $request.remoteAddress(),
                  uri = $request.uri, code = exc.code,
                  path_params = pathParams, query_params = queryParams,
                  content_body = optBody, error_msg = $exc.msg
            return await request.respond(exc.code)
          except CatchableError as exc:
            warn "Unexpected error occured while processing a request",
                  meth = $request.meth, peer = $request.remoteAddress(),
                  uri = $request.uri, path_params = pathParams,
                  query_params = queryParams, content_body = optBody,
                  error_msg = $exc.msg, error_name = $exc.name
            return await request.respond(Http503)
```

Reply:
Ok, good to know, no need to change anything then :)